### PR TITLE
demos: Vulkaninfo add help flag

### PR DESCRIPTION
**Pull request by shannon-lunarg**
Vulkaninfo prints usage information when the "-h" or "--help" option.

Change-Id: I43886920e5ce1feb2e293ed46a3e8c23a217ca69

### CODE REVIEW

File: demos/vulkaninfo.c [hunk 1]
```diff
@@ -723,7 +723,7 @@ void PrintJsonHeader(const int vulkan_major, const int vulkan_minor, const int v
     printf("{\n");
     printf("\t\"$schema\": \"https://schema.khronos.org/vulkan/devsim_1_0_0.json#\",\n");
     printf("\t\"comments\": {\n");
-    printf("\t\t\"desc\": \"JSON configuration file describing GPU %u. Generated using the VulkanInfo program.\",\n", selected_gpu);
+    printf("\t\t\"desc\": \"JSON configuration file describing GPU %u. Generated using the Vulkaninfo program.\",\n", selected_gpu);
```
Comment (mikew-lunarg) [inline demos/vulkaninfo.c]: I believe the executable file is called "vulkaninfo" all lower-case.  Perhaps just use that for consistency.
Or printf using argv[0], which would identify the actual binary invoked.

File: demos/vulkaninfo.c [hunk 2]
```diff
@@ -3092,6 +3092,23 @@ static void ConsoleEnlarge() {
...
 
+void print_usage(char* argv0) {
+    printf("\nVulkaninfo - Summarize Vulkan information in relation to the current environment.\n\n");
+    printf("USAGE: %s [options]\n\n", argv0);
+    printf("OPTIONS:\n");
+    printf("-h, --help                        Print this help.\n");
+    printf("--html                            Produce an html version of Vulkaninfo output,\n");
+    printf("                                  saved as \"vulkaninfo.html\" in the directory in\n");
+    printf("                                  which the command is run.\n");
+    printf("-j, --json, --json=<gpu-number>   Produce a json version of Vulkaninfo output\n");
```
Comment (mikew-lunarg) [inline demos/vulkaninfo.c]: suggest splitting this into two separate lines: the first w/o gpu-number (and identify which gpu when not specified), the second with the gpu specifier.  That will allow reduced indentation.

File: demos/vulkaninfo.c [hunk 3]
```diff
@@ -3092,6 +3092,23 @@ static void ConsoleEnlarge() {
 }
 #endif
 
+void print_usage(char* argv0) {
+    printf("\nVulkaninfo - Summarize Vulkan information in relation to the current environment.\n\n");
```
Comment (mikew-lunarg) [inline demos/vulkaninfo.c]: suggest "vulkaninfo" all lower-case; see other comment.  will be consistent w vulkaninfo.html

File: demos/vulkaninfo.c [hunk 4]
```diff
@@ -3115,7 +3132,10 @@ int main(int argc, char **argv) {
     const uint32_t vulkan_patch = VK_VERSION_PATCH(VK_HEADER_VERSION);
 
     for (int i = 1; i < argc; i++) {
-        if (strcmp(argv[i], "--html") == 0) {
+        if (strcmp(argv[i], "--help") == 0 || strcmp(argv[i], "-h") == 0) {
```
Comment (mikew-lunarg) [inline demos/vulkaninfo.c]: Is usage also printed if a switch is not recognized?

File: demos/vulkaninfo.c [hunk 5]
```diff
@@ -3115,7 +3132,10 @@ int main(int argc, char **argv) {
...
+            print_usage(argv[0]);
+            return 0;
```
Comment (mikew-lunarg) [inline demos/vulkaninfo.c]: Usage often returns non-zero, as it is not really a successful invocation of the tool

**Comment by shannon-lunarg** [comment]
The changes @mikew-lunarg suggested have been made. Unless there are any additional suggestions I will merge this once it passes all tests.

**Comment by mark-lunarg** [comment]
Shannon -- clang-format and commit message fixups needed, push when you CI passes.

**Comment by shannon-lunarg** [comment]
@mark-lunarg, I will address the clang format issues but as far as I can tell my two commit messages are fine. What I see on Travis is a problem with the length of "layers: remove duplicate GetEnabledExtensions() function" but this commit is a few commits back. Am I supposed to change someone else's commit message?

**Comment by mark-lunarg** [comment]
No, false alarm!

**Pull request merged**